src/batchUploader.ts

```diff
@@ -129,21 +130,70 @@ export class BatchUploader {
         return offlineStoragePercentage >= rampNumber;
     }
 
+    // debounce AST just in case multiple events are fired in a short period of time due to browser differences
+    private shouldDebounceAndUpdateLastASTTime(): boolean {
+        const now = Date.now();
+        if (now - this.lastASTEventTime < this.AST_DEBOUNCE_MS) {
+            return true;
+        }
+
+        this.lastASTEventTime = now;
+        return false;
+    }
+
+    // https://go.mparticle.com/work/SQDSDKS-7133
+    private createBackgroundASTEvent(): SDKEvent {
+        const now = Date.now();
+        const { _Store, Identity, _timeOnSiteTimer  } = this.mpInstance;
+        const { sessionId, deviceId, sessionStartDate, SDKConfig } = _Store;
+        return {
+            EventDataType: MessageType.AppStateTransition,
+            EventName: 'Application Exit',
+            EventCategory: EventType.Other,
+            Timestamp: now,
+            SessionId: sessionId,
+            MPID: Identity.getCurrentUser()?.getMPID(),
+            DeviceId: deviceId,
+            IsFirstRun: false,
+            SourceMessageId: '',
```

Comment:
I think this should use `utils.generateUniqueId` to generate an ID.

Reply:
good catch